.github/workflows/legacy-os.yml

```diff
@@ -16,3 +16,13 @@ jobs:
         run: |
           { { ! sudo scripts/lib/upgrade-zulip-stage-2 2>&1 >&3; } | tee upgrade.err; } 3>&1 >&2
           grep -Fq 'upgrade-zulip-stage-2: Unsupported platform: ubuntu 16.04' upgrade.err
+
+      - name: Report Status
+        if: always()
+        uses: amanagr/notify-zulip-action@master
+        with:
+          status: ${{ job.status }}
+          notify_when: "failure"
```

Comment:
Can this not just be [`if: ${{ failure() }}`](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#failure)?